### DISCUSSION TITLE
Handling TIFF images

### DISCUSSION THREAD

**Discussion by Synchro** [spatie/laravel-medialibrary#2794] [Q&A]
I'm using imagemagick as my driver, which has TIFF format support, but I can't make it work in MLP. Is there something I can to do in order to make TIFF format work with MLP, ML, and the underlying image and glide libraries?

I've found that [switching to imagick is all that is required to have TIFF work in Glide](https://github.com/thephpleague/glide/issues/149), so that's not what's stopping it.

**Comment by freekmurze** [comment]
I think we should add a check for the driver here: https://github.com/spatie/laravel-medialibrary/blob/caa4e9b7bc3e2775dc0ee3c61a87e7030b2d73ae/src/Conversions/ImageGenerators/Image.php#L20-L28

If Imagick is detected, the necessary values for TIFF could be added.

Feel free to try this out locally and send a PR if it all works. Also, add a test to make ensure that it works.

**Reply by Synchro** [reply]
I've been messing with some experiments and discovered [an underlying bug in Glide that stops TIFF from working](https://github.com/thephpleague/glide/issues/344); I've submitted a PR for that, but it will ultimately mean we need to bump a dep version when it makes its way to a release.

This also needs to be propagated through spatie/image, and I'm doing a driver check in [the fork I'm working on for that too](https://github.com/Synchro/image). Eventually I'll need to add the same checks in ML as well – quite the daisy chain!

I think there might be a bug in a spatie/image test that fails if imagick is present – I think Intervention may default to imagick if it's present, and the tests are not expecting that. It might just be something I've broken though.

Regarding those format checks, isn't that something that should be delegated to spatie/image so we don't have to repeat them, here? i.e. shouldn't ML support whatever image formats spatie/image supports, rather than introducing its own limits?
